mlir-tensorrt/test/python/IntegrationTests/test_non_dps_cconv.py

```diff
@@ -0,0 +1,252 @@
+        if debug:
+            opts.set_debug_options(False, [], "tmp")
+        exe = compiler.compiler_stablehlo_to_executable(client, m.operation, opts)
+        return exe
+
+
+def test_single_return():
+    exe = compile_executable(single_return)
+    session_options = runtime.RuntimeSessionOptions(num_devices=1, device_id=0)
+    session = runtime.RuntimeSession(session_options, exe)
+    arg0 = client.create_memref(
+        np.arange(0.0, 24.0, dtype=np.float32).reshape(2, 3, 4).data,
+        device=devices[0],
+        stream=stream,
+    )
+    results = session.execute_function(
+        "main", in_args=[arg0], stream=stream, client=client
+    )
+
+    output = np.asarray(client.copy_to_host(results[0], stream=stream))
+    stream.sync()
+
+    print(output)
+
+
+def test_multiple_return():
+    exe = compile_executable(multiple_return)
+    session_options = runtime.RuntimeSessionOptions(num_devices=1, device_id=0)
+    session = runtime.RuntimeSession(session_options, exe)
+    arg0 = client.create_memref(
+        np.arange(0.0, 24.0, dtype=np.float32).reshape(2, 3, 4).data,
+        device=devices[0],
+        stream=stream,
+    )
+    results = session.execute_function(
+        "main", in_args=[arg0], stream=stream, client=client
+    )
+
+    output_0 = np.asarray(client.copy_to_host(results[0], stream=stream))
+    output_1 = np.asarray(client.copy_to_host(results[1], stream=stream))
+
+    stream.sync()
+
+    print(output_0)
+    print(output_1)
+
+
+def test_dynamic_shape():
+    exe = compile_executable(dynamic_shape)
+    session_options = runtime.RuntimeSessionOptions(num_devices=1, device_id=0)
+    session = runtime.RuntimeSession(session_options, exe)
+    arg0 = client.create_memref(
+        np.arange(0.0, 8.0, dtype=np.float32).reshape((4, 2)).data,
+        device=devices[0],
+        stream=stream,
+    )
+    arg1 = client.create_memref(
+        np.ones((4, 2), dtype=np.float32).data, device=devices[0], stream=stream
+    )
+
+    results = session.execute_function(
+        "main", in_args=[arg0, arg1], stream=stream, client=client
+    )
+
+    output = np.asarray(client.copy_to_host(results[0], stream=stream))
+    stream.sync()
+
+    print(output)
+
+
+def test_session_tracking_d2h():
+    exe = compile_executable(session_tracking_h2h)
+    session_options = runtime.RuntimeSessionOptions(num_devices=1, device_id=0)
+    session = runtime.RuntimeSession(session_options, exe)
+    results = session.execute_function("main", in_args=[], stream=stream, client=client)
+    stream.sync()
+    print(np.asarray(results[0]))
+
+
+def test_empty_shape_tensor():
+    exe = compile_executable(empty_shape_tensor)
+    session_options = runtime.RuntimeSessionOptions(num_devices=1, device_id=0)
+    session = runtime.RuntimeSession(session_options, exe)
+    results = session.execute_function("main", in_args=[], stream=stream, client=client)
+    stream.sync()
+    print(np.asarray(results[0]))
+
```

Comment:
Would be good to get some test coverage for scalar results and mixed memref/scalar results as well.

Reply:
I have added tests to cover the scalar and mixed results.